packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -1400,7 +1400,13 @@ class SliverConstrainedCrossAxis extends SingleChildRenderObjectWidget {
 
   @override
   RenderSliverConstrainedCrossAxis createRenderObject(BuildContext context) {
-    return RenderSliverConstrainedCrossAxis(maxExtent: maxExtent);
+    final RenderSliverConstrainedCrossAxis renderObject = RenderSliverConstrainedCrossAxis(maxExtent: maxExtent);
+    // Typically, parent data is setup in the parent via a call to [RenderObject.setupParentData] but we must
+    // do this manually here to not have the default flex value of 1.
+    final SliverPhysicalContainerParentData renderObjectParentData = SliverPhysicalContainerParentData();
```

Comment:
The docs on parentData say `The parent data field must not be directly set, except by calling [setupParentData] on the parent node.`. 
This is probably not something we want to do. Instead, should slivers that aren't flexing just be laid out, and if they exceed the extent provide an error message that recommends the cross axis extent sliver?

Reply:
Just refactored the code so that `SliverConstrainedCrossAxis` is a combination of a parent data widget and a render object widget, and this works perfectly. I'm still feeling a bit iffy on putting `crossAxisFlex` in `SliverPhysicalParentData` since `crossAxisFlex` is only really used in `SliverCrossAxisGroup`, but I think this is ok.

Reply:
Refactoring the SliverCrossAxisExtent should probably be in its own PR, as well as the changes to SliverGeometry. I am about to take another pass on this, but that's my first thought